walk.go

```diff
@@ -79,10 +79,10 @@ func walkiter(s *queue, walkFn filepath.WalkFunc) (haderror error) {
 				haderror = walkFn(ourname, info, err)
 			}
 			file.Close()
-			for i := len(names) - 1; i >= 0; i-- {
+			for _, name := range names {
```

Comment:
There is a reason the old code is like this: on OSX the directory entries are given in reverse order. it doesn't matter much on linux, but it's a nice touch to have on OSX. A better solution would be to sort the (presumably small) list of files before traversing them.